common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -111,23 +112,19 @@ def may_certify_for_course(
             course's certificates before the course has ended.
         has_ended (bool): Whether the course has ended.
         certificate_available_date (datetime): the date the certificate is available on for the course.
+        self_paced (bool): Whether the course is self-paced.
     """
     show_early = (
         certificates_display_behavior in ('early_with_info', 'early_no_info')
         or certificates_show_before_end
     )
-    past_availability_date = (
+    past_available_date = (
         certificate_available_date
         and certificate_available_date < datetime.now(utc)
     )
+    ended_without_available_date = (certificate_available_date is None) and has_ended
```

Comment:
out of curiosity could this be written as `has_ended and certificate_available_date is None` without parentheses? Or does that see if `(has_ended and certificate_available_date)` is `None`?

Reply:
https://docs.python.org/3/reference/expressions.html#operator-precedence
`is` (or any comparison) has higher precedence than boolean AND, so we could eliminate the parens and put it in either order.  I tend to use parentheses even when I don't have to, so I don't have to think too hard.